modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheMapEntry.java

```diff
@@ -5729,19 +5732,28 @@ private UpdateClosure(
                 }
             }
 
-            if (val != null) {
-                newRow = entry.cctx.offheap().dataStore(entry.localPartition()).createRow(
-                    entry.cctx,
-                    entry.key,
-                    val,
-                    ver,
-                    expireTime,
-                    oldRow);
-            }
-            else {
-                // Create tombstone for preloaded removal.
-                newRow = entry.cctx.offheap().dataStore(entry.localPartition()).createRow(entry.cctx, entry.key(),
-                    TombstoneCacheObject.INSTANCE, ver, entry.cctx.shared().ttl().tombstoneExpireTime(), oldRow);
+            while (true) {
```

Comment:
Do we need the same wrapping somewhere else? `AtomicCacheUpdateClosure#update()` for example.